components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -11048,12 +11048,24 @@ public static boolean isRecommendationEnabled(String tenantDomain) {
         return false;
     }
 
-    public static void publishEvent(EventPublisherType type, EventPublisherEvent event, String errorMessage) {
+    public static void addNewEventPublisher(EventPublisherType type) {
+        try {
+            eventPublishers.putIfAbsent(type, eventPublisherFactory.getEventPublisher(type));
```

Comment:
We need to make sure, the factories support any arbitrary type.

Reply:
Will support it from default flow